> Makeconf is included into the Makefile so you should be able to set PREFIX, BLAS and LAPACK in makeFlags.

Yea that would have been nicer, but apparently the flags are not passed to ld:

```
extra flags after to /nix/store/p792j5f44l3f0xi7ai5jllwnxqwnka88-binutils-2.31.1/bin/ld:
  -rpath
  /nix/store/9df65igwjmf2wbw0gbrrgair6piqjgmi-glibc-2.31/lib
  -rpath
  /nix/store/z5g0y84g2iknwwgfhw9wslbbzgw1w22k-gfortran-9.3.0-lib/lib
/nix/store/p792j5f44l3f0xi7ai5jllwnxqwnka88-binutils-2.31.1/bin/ld: /build/ccOyKBv6.o: in function `ztest_':
tqr1up.f:(.text+0x18d): undefined reference to `zgerc_'
/nix/store/p792j5f44l3f0xi7ai5jllwnxqwnka88-binutils-2.31.1/bin/ld: /build/ccOyKBv6.o: in function `ctest_':
```

Though it does work if I use `NIX_LDFLAGS`. Thanks for the idea.

--- a/pkgs/development/libraries/qrupdate/default.nix
+++ b/pkgs/development/libraries/qrupdate/default.nix
@@ -5,24 +5,33 @@
 , lapack
 , which
 }:
-stdenv.mkDerivation {
-  name = "qrupdate-1.1.2";
+
+stdenv.mkDerivation rec {
+  pname = "qrupdate";
+  version = "1.1.2";
   src = fetchurl {
-    url = "mirror://sourceforge/qrupdate/qrupdate-1.1.2.tar.gz";
+    url = "mirror://sourceforge/qrupdate/${pname}-${version}.tar.gz";
     sha256 = "024f601685phcm1pg8lhif3lpy5j9j0k6n0r46743g4fvh8wg8g2";
   };
 
-  configurePhase =
-    ''
-      export PREFIX=$out
-      sed -i -e 's,^BLAS=.*,BLAS=-L${blas}/lib -lblas,' \
-          -e 's,^LAPACK=.*,LAPACK=-L${lapack}/lib -llapack,' \
-          Makeconf
-    ''
-    + stdenv.lib.optionalString blas.isILP64
-    ''
-      sed -i Makeconf -e '/^FFLAGS=.*/ s/$/-fdefault-integer-8/'
-    '';
+  preBuild =
+    # Check that blas and lapack are compatible
+    assert (blas.isILP64 == lapack.isILP64);
+  # We don't have structuredAttrs yet implemented, and we need to use space
+  # seprated values in makeFlags, so only this works.
+  ''
+    makeFlagsArray+=(
+      "LAPACK=-L${lapack}/lib -llapack"
+      "BLAS=-L${blas}/lib -lblas"
+      "PREFIX=${placeholder "out"}"
+      ${stdenv.lib.optionalString blas.isILP64
+      # Use their FFLAGS along with `-fdefault-integer-8`. If another
+      # application intends to use arpack, it should add this to it's FFLAGS as
+      # well. Otherwise (e.g): https://savannah.gnu.org/bugs/?50339
+      "FFLAGS=-fimplicit-none -O3 -funroll-loops -fdefault-integer-8"
+      }
+    )
+  '';
 
   doCheck = true;
 
@@ -32,14 +41,14 @@ stdenv.mkDerivation {
 
   installTargets = stdenv.lib.optionals stdenv.isDarwin [ "install-staticlib" "install-shlib" ];
 
-  buildInputs = [ gfortran blas lapack ];
+  buildInputs = [ gfortran ];
 
   nativeBuildInputs = [ which ];
 
   meta = with stdenv.lib; {
     description = "Library for fast updating of qr and cholesky decompositions";
     homepage = "https://sourceforge.net/projects/qrupdate/";
-    license = licenses.gpl3;
+    license = licenses.gpl3Plus;
     maintainers = with maintainers; [ doronbehar ];
     platforms = platforms.unix;
   };

